rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1145,24 +1123,63 @@ private void processPegoutRequests(Transaction rskTx) {
         }
     }
 
-    private void addToPegoutsWaitingForConfirmations(
-        BtcTransaction generatedTransaction,
-        PegoutsWaitingForConfirmations pegoutWaitingForConfirmations,
-        Keccak256 pegoutCreationRskTxHash,
-        Coin amount
-    ) {
-        if (activations.isActive(ConsensusRule.RSKIP146)) {
-            // Add the TX
-            pegoutWaitingForConfirmations.add(generatedTransaction, rskExecutionBlock.getNumber(), pegoutCreationRskTxHash);
-            // For a short time period, there could be items in the pegout request queue that don't have the pegoutCreationRskTxHash
-            // (these are pegouts created right before the consensus rule activation, that weren't processed before its activation)
-            // We shouldn't generate the event for those pegouts
-            if (pegoutCreationRskTxHash != null) {
-                eventLogger.logReleaseBtcRequested(pegoutCreationRskTxHash.getBytes(), generatedTransaction, amount);
-            }
-        } else {
-            pegoutWaitingForConfirmations.add(generatedTransaction, rskExecutionBlock.getNumber());
+    private void settleReleaseRequest(PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations, BtcTransaction pegoutTransaction, Keccak256 releaseCreationTxHash, Coin requestedAmount) {
```

Comment:
I don't think we need to pass around the `pegoutsWaitingForConfirmations`, we can simply get it from the storageProvider right?

Reply:
yes, thats right, but the `getPegoutsWaitingForConfirmations` could throw an `IOException` that has to be handled all over the code. See for example in [processPegoutsInBatch](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1205) and [processPegoutsIndividually](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1166), that the map is an input of the method already handled in [processPegoutRequests](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1120) method.

I thought it was better to do it this way to have less impact. But let me know what you think!

Reply:
Yeah, no big deal. Thanks

Reply:
thanks to you :)